cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.LocalObjectReference;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentStrategy;
+import io.fabric8.kubernetes.api.model.apps.DeploymentStrategyBuilder;
+import io.fabric8.kubernetes.api.model.apps.RollingUpdateDeploymentBuilder;
+import io.strimzi.api.kafka.model.JmxTransResources;
+import io.strimzi.api.kafka.model.JmxTransSpec;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.api.kafka.model.KafkaJmxAuthenticationPassword;
+import io.strimzi.api.kafka.model.KafkaJmxOptions;
+import io.strimzi.api.kafka.model.Probe;
+import io.strimzi.api.kafka.model.ProbeBuilder;
+import io.strimzi.api.kafka.model.template.JmxTransOutputDefinitionTemplate;
+import io.strimzi.api.kafka.model.template.JmxTransQueryTemplate;
+import io.strimzi.operator.common.model.Labels;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+/*
+ * Class for handling JmxTrans configuration passed by the user. Used to get the resources needed to create the
+ * JmxTrans deployment including: config map, deployment, and service accounts.
+ */
+public class JmxTrans extends AbstractModel {
+
+    // Configuration defaults
+    private static final int DEFAULT_HEALTHCHECK_DELAY = 15;
```

Comment:
We seem to have this twice ... here and in `JmxTransSpec`. Can we have it only once in the code?

Reply:
Thank you for the spot, will change it!

Reply:
I guess that is fair point, but maybe we can do it better here? I don't think there is any actual reason for it being twice for other deployments.

Reply:
I guess we could just hard code it because these are only used once?